packages/core/core/src/applyRuntimes.js

```diff
@@ -66,8 +69,23 @@ export default async function applyRuntimes({
         if (applied) {
           let runtimeAssets = Array.isArray(applied) ? applied : [applied];
           for (let {code, dependency, filePath, isEntry} of runtimeAssets) {
+            let sourceName = path.join(
```

Comment:
Maybe runtimes shouldn't return a filePath but a directory? Otherwise people might be confused that the asset doesn't actually use the filePath they specify?

Reply:
for the sourcemap we do need some filepath, not sure what we should do for the actual asset it returns though

Reply:
Yeah I mean core would still add a filename but plugins would only return a dir. right now they return a file but the filename part isn't used

Reply:
So we should just make core append /index or something to the dirname as filename?